portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java

```diff
@@ -840,6 +844,14 @@ public void deleteLayout(
 
 		layout = delete(layout);
 
+		// User preferences for customizable pages
+
+		if (layout.isCustomizable()) {
```

Comment:
@ChrisKian - something to think about: maybe this check is unnecessary? With this check, user preferences would not be deleted if a page was modified to be not customizable before it was deleted. And looking at this again, I don't see any downside to running the _removeUserPreferences method for all pages (besides performance considerations).